alphabase/io/tempmmap.py

```diff
@@ -92,7 +92,59 @@ def array(shape: tuple, dtype: np.dtype) -> np.ndarray:
             offset=offset
         ).reshape(shape)
 
-def mmap_array(hdf_file: str) -> np.ndarray:
+def create_empty_mmap(shape: tuple, dtype: np.dtype, path: str = None, overwrite = False):
+    """Initialize a new HDF5 file compatible with mmap. Returns the path to the initialized file.
+    File can be mapped using the mmap_array_from_path function.
+    
+    Parameters
+    ----------
+    shape : tuple
+        A tuple with the shape of the array.
+    dtype : type
+        The np.dtype of the array.
+    path : str, optional
+        The path to the file that should be created.
+        Defaults to None.
+        If None a random file name will be generated.
+    overwrite : bool , optional
+        If True the file will be overwritten if it already exists.
+        Defaults to False.
+
+    Returns
+    -------
+    str
+        path to the newly created file.
+    """
+    
+    #if path does not exist generate a random file name in the TEMP directory
+    if path is None:
+        temp_file_name = os.path.join(
+            TEMP_DIR_NAME,
+            f"temp_mmap_{np.random.randint(2**63)}.hdf"
+        )
+    else:
+        #check that if overwrite is false the file does not already exist
+        if not overwrite:
+            if os.path.exists(path):
+                raise ValueError("The file already exists. Set overwrite to True to overwrite the file or choose a different name.")
+        if not os.path.basename.endswith('.hdf'):
+            raise ValueError("The chosen file name needs to end with .hdf")
+        if os.path.isdir(os.path.commonpath(path)):
+            temp_file_name = path
+        else:
+            raise ValueError("The directory in which the file should be created does not exist.")
+        
+    with h5py.File(temp_file_name, "w") as hdf_file:
+        array = hdf_file.create_dataset(
+            "array",
+            shape=shape,
+            dtype=dtype
+        )
+        array[0] = 0
+
+    return temp_file_name
+
+def mmap_array_from_path(hdf_file: str) -> np.ndarray:
```

Comment:
Yes, this name is better than mmap_array

Reply:
agree :) 
I am currently using it to perform multi-threaded operations that write results to the same mmap object concurrently and its working perfectly. Interestingly, using this setup I don't encounter issues where the HDF5 file becomes locked when I have a writable instance as I do using base h5py.

Reply:
I think HDF5 locks is required for SLURM, multi-threading (not multi-processing) is fine